internal/interpreter/global_component.rs

```diff
@@ -107,28 +110,27 @@ pub fn instantiate(description: &CompiledGlobal) -> (String, Pin<Rc<dyn GlobalCo
                     }
                 }
             }
-            let g = i_slint_backend_selector::NativeGlobals::instantiate(
+            i_slint_backend_selector::NativeGlobals::instantiate(
                 element.native_class.class_name.as_ref(),
-            );
-            (name.clone(), g)
+            )
         }
         CompiledGlobal::Component { component, .. } => {
             generativity::make_guard!(guard);
             let description = component.unerase(guard);
-            let component = &description.original;
-            let g = Rc::pin(GlobalComponentInstance(crate::dynamic_component::instantiate(
+            Rc::pin(GlobalComponentInstance(crate::dynamic_component::instantiate(
                 description.clone(),
                 None,
                 None,
-            )));
-            let id = if component.is_global() {
-                component.root_element.borrow().id.clone()
-            } else {
-                component.id.clone()
-            };
-            (id, g)
+                globals.clone(),
```

Comment:
This clones the entire HashMap. Maybe it's worth putting that into an RC?

Reply:
a `Rc<RefCell<HashMap<String, Rc<_>>`
Well no that won't work. there would be cycle.

I thought about trying to take the Hashmap and then take it back, but then i thought there is no point. The hashmap is containing Rc anyway.

Yes, that means this use O(n²) memory in the number of globals.  But there should not be that many globals anyway.

Reply:
True true, there shouldn't be many and this design is much simpler.